src/jit/flowgraph.cpp

```diff
@@ -4222,6 +4222,16 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr,
         compInlineResult->NoteBool(InlineObservation::CALLEE_IS_FORCE_INLINE, isForceInline);
         compInlineResult->NoteInt(InlineObservation::CALLEE_IL_CODE_SIZE, codeSize);
 
+#ifdef DEBUG
+
+        // If inlining, this method should still be a candidate.
+        if (compIsForInlining())
+        {
+            assert(compInlineResult->IsCandidate());
+        }
```

Comment:
I had it that way, but changed it. 

I'm not really a fan of compound conditions in asserts (especially so when calls are involved), since it's more pain to untangle which part went wrong when debugging.